### ISSUE TITLE
Give hints to the automatic tools (linters, fixers)

### ISSUE THREAD

**Issue by severo** [georchestra/georchestra#2258]
In order to enforce a common coding style, at least for some of the languages (JS, Java, markdown), we could add the following files at the georchestra root directory:
- .eslintrc
- .prettierrc
- .prettierrcignore
etc.

At least create them in order to disable them and avoid that global settings in the developer system might modify indentation and pollute the commits if done without care.

**Comment by severo** [comment]
Note that we can set a different file in every module, as generally the linters will look for their configuration file in the filesystem tree above their current location.

**Comment by fvanderbiest** [comment]
OK, feel free to provide these config files with your future PRs, on a module basis, as you go...

**Comment by severo** [comment]
To take into account: https://help.github.com/articles/dealing-with-line-endings/

**Comment by groldan** [comment]
Hi, I see `.eslintrc` is for JavaScript?

For reference, for Java GeoTools uses the [fmt-maven-plugin](https://github.com/coveo/fmt-maven-plugin), see [here](https://github.com/geotools/geotools/blob/faec543710222ff23145defa47935ffa6e7e6219/pom.xml#L1946)

It looks like there're [plugins for IntelliJ and Eclipse](https://github.com/google/google-java-format) but I haven't tried it yet.

**Comment by severo** [comment]
I think we could look at https://support.codacy.com/hc/en-us/articles/207994335-Code-Patterns and use it as a base for checking the code styles.

I don't like very much the idea of a maven plugin that automatically modify the code. I would largely prefer configurations files that editors (Eclipse, IntelliJ, Atom, vim, etc.) could use to provide insights to the developer. And that would allow Codacy to check and give a OK/NOK on the PR.

**Comment by severo** [comment]
> Not sure we want a single linter for the whole javascript part. (old ExtJS for viewer vs console for example).

You're right. ESLint and others look for the file with highest priority in a given lookup order, which allows to choose a specific code style for a given module or directory.
